Tasks/Xcode/xcodeutils.ts

```diff
@@ -46,6 +46,52 @@ export function findDeveloperDir(xcodeVersion: string): string {
     return discoveredDeveloperDir;
 }
 
+export function buildDestinationArgs(platform: string, devices: string[], targetingSimulators: boolean): string[] {
+    let destinations: string[] = [];
+
+    devices.forEach((device: string) => {
+        device = device.trim();
+
+        let destination;
+        if (device) {
+            if (targetingSimulators) {
+                destination = `platform=${platform} Simulator`;
+            }
+            else {
+                destination = `platform=${platform}`;
+            }
+
+            let matches = device.match(/([0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12})/);
+            if (matches) {
+                destination += `,id=${matches[0]}`;
+            }
```

Comment:
It would be better to just take the string the user provided and be able to pass it to the tool runner. Can we avoid doing all this parsing?

Reply:
Done. Now you specify "iPhone X,OS=11.1" instead of "iPhone X (11.1)". That's probably clearer given all the iPads with parenthetical sizes. We can use a multi-line control if we want to accept multiple devices.